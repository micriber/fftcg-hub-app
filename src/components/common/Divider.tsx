import React from 'react';
import {StyleSheet, View, ViewStyle} from 'react-native';

type Props = {
  style: ViewStyle;
};

const Divider = ({style}: Props) => {
  return <View style={[styles.divider, style]} />;
};

const styles = StyleSheet.create({
  divider: {
    borderBottomColor: '#000000',
    borderBottomWidth: StyleSheet.hairlineWidth,
    padding: 1,
    width: '100%',
  },
});

export default Divider;
