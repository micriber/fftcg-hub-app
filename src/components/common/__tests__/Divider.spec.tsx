/**
 * @format
 */

import 'react-native';
import React from 'react';
import Divider from '../Divider';
import {render} from '@testing-library/react-native';

// More about testing in this repository:
// https://github.com/bamlab/react-native-testing/blob/master/src/pages/Home/__tests__/Home.test.tsx
describe('[Component] Divider', () => {
  it('renders correctly', async () => {
    const root = render(<Divider />);

    expect(root.toJSON()).toMatchSnapshot();
  });
});
