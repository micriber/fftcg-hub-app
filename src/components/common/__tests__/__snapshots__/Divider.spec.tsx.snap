// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] Divider renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "borderBottomColor": "#000000",
        "borderBottomWidth": 0.5,
        "padding": 1,
        "width": "100%",
      },
      undefined,
    ]
  }
/>
`;
